packages/cubejs-server-core/src/core/CompilerApi.js

```diff
@@ -33,7 +33,7 @@ export class CompilerApi {
     return this.graphqlSchema;
   }
 
-  async getCompilers({ requestId } = {}) {
+  async getCompilers({ requestId, skipQueryFactory = false } = {}) {
```

Comment:
we should skip queryFactory because inside it run driverFactory for all dataSources. 
And we need to check it one by one for security context